lib/types/src/memory_view.rs

```diff
@@ -64,6 +67,44 @@ where
             _phantom: PhantomData,
         }
     }
+
+    /// Creates a subarray view from this MemoryView.
+    pub fn subarray(&self, range: impl RangeBounds<usize>) -> Self {
+        let start: usize = match range.start_bound() {
+            Bound::Unbounded => 0,
+            Bound::Included(start) => *start,
+            Bound::Excluded(start) => *start + 1,
+        };
+        let end: usize = match range.end_bound() {
+            Bound::Unbounded => self.length,
+            Bound::Included(end) => *end,
+            Bound::Excluded(end) => *end - 1,
+        };
+        Self {
+            ptr: unsafe { self.ptr.add(start) },
+            length: (end - start),
+            _phantom: PhantomData,
+        }
+    }
+
+    /// Copy the contents of the source slice into this `MemoryView`.
+    ///
+    /// This function will efficiently copy the memory from within the wasm
+    /// module’s own linear memory to this typed array.
+    ///
+    /// # Safety
+    ///
+    /// This method is unsafe because the caller will need to make sure
+    /// there are no data races when copying memory into the view.
+    pub unsafe fn copy_from(&self, src: &[T]) {
+        assert!(
+            src.len() == self.length,
```

Comment:
If we change for `src.len() >= self.length`, we could be able to write from a slice that is smaller than the memory view. It can be useful. Ideally, we should also support an offset. I don't know what's your idea here: Do we want to provide a richer API, or just a super basic one?

Reply:
Fixed, I actually remove the assertion. If the `src.len() < self.length` we actually just copy it's totality, and if it's bigger we get the smaller slice.